types/powerbi-visuals-tools/index.d.ts

```diff
@@ -0,0 +1,1372 @@
+    /** Represents evaluated, named, custom objects in a DataView. */
+    interface DataViewObjects {
+        [name: string]: DataViewObject;
+    }
+
+    /** Represents an object (name-value pairs) in a DataView. */
+    interface DataViewObject {
+        /** Map of property name to property value. */
+        [propertyName: string]: DataViewPropertyValue | DataViewObjectMap;
+
+        /** Instances of this object. When there are multiple instances with the same object name they will appear here. */
+        // $instances?: DataViewObjectMap; - moved to indexed property as optional type
+    }
+
+    interface DataViewObjectWithId {
+        id: string;
+        object: DataViewObject;
+    }
+
+    interface DataViewObjectPropertyIdentifier {
+        objectName: string;
+        propertyName: string;
+    }
+
+    type DataViewObjectMap = { [id: string]: DataViewObject };
+
+    type DataViewPropertyValue = PrimitiveValue | StructuralObjectValue;
+}
+
+declare namespace powerbi.data {
+    /** Defines a match against all instances of given roles. */
+    interface DataViewRoleWildcard {
+        kind: DataRepetitionKind.RoleWildcard;
+        roles: string[];
+        key: string;
+    }
+}
+
+declare namespace powerbi {
+    /** Encapsulates the identity of a data scope in a DataView. */
+    interface DataViewScopeIdentity {
+        kind: DataRepetitionKind.ScopeIdentity;
+
+        /** Predicate expression that identifies the scope. */
+        expr: data.ISQExpr;
+
+        /** Key string that identifies the DataViewScopeIdentity to a string, which can be used for equality comparison. */
+        key: string;
+    }
+}
+
+declare namespace powerbi.data {
+    /** Defines a match against all instances of a given DataView scope. Does not match Subtotals. */
+    interface DataViewScopeWildcard {
+        kind: DataRepetitionKind.ScopeWildcard;
+        exprs: ISQExpr[];
+        key: string;
+    }
+}
+
+declare namespace powerbi.data {
+    import IStringResourceProvider = jsCommon.IStringResourceProvider;
+
+    type DisplayNameGetter = ((resourceProvider: IStringResourceProvider) => string) | string;
+}
+
+declare namespace powerbi.data {
+    /** Defines a selector for content, including data-, metadata, and user-defined repetition. */
+    interface Selector {
+        /** Data-bound repetition selection. */
+        data?: DataRepetitionSelector[];
+
+        /** Metadata-bound repetition selection.  Refers to a DataViewMetadataColumn queryName. */
+        metadata?: string;
+
+        /** User-defined repetition selection. */
+        id?: string;
+    }
+
+    type DataRepetitionSelector =
+        DataViewScopeIdentity |
+        DataViewScopeWildcard |
+        DataViewRoleWildcard |
+        DataViewScopeTotal;
+
+	interface SelectorsByColumn { }
+}
+
+declare namespace powerbi.data {
+    // intentionally blank interfaces since this is not part of the public API
+
+    interface ISemanticFilter { }
```

Comment:
Can't stress enough that you should not ever do this. The `no empty interfaces` rule is there to save you, not to make your life harder. This interface being empty means that `StructuralObjectValue` behaves as `{}` - literally anything is assignable to it!